driver-core/src/test/java/com/datastax/driver/core/policies/DowngradingConsistencyRetryPolicyIntegrationTest.java

```diff
@@ -15,90 +15,338 @@
+        return new Object[][]{
+            {WriteTypePrime.SIMPLE},
+            {WriteTypePrime.BATCH},
+            {WriteTypePrime.COUNTER},
+            {WriteTypePrime.CAS}
+        };
+    }
+
+
+    /**
+     * @return Write Types for which we expect an ignore if used and there are received acks.
+     */
+    @DataProvider
+    public static Object[][] ignoreWriteTypesWithReceivedAcks() {
+        return new Object[][] {
+            {WriteTypePrime.SIMPLE},
+            {WriteTypePrime.BATCH}
+        };
+    }
+
+
+    @Test(groups = "short", dataProvider = "consistencyLevels")
+    public void should_retry_once_on_same_host_with_reduced_consistency_level_on_read_timeout(int received, ConsistencyLevel expectedCL) {
+        simulateError(1, read_request_timeout, new ReadTimeoutConfig(received, received+1, true));
 
-    @Test(groups = "long")
-    public void should_downgrade_if_not_enough_replicas_for_requested_CL() {
-        CCMBridge ccm = null;
-        Cluster cluster = null;
         try {
-            // 3-node cluster, keyspace with RF = 3
-            ccm = CCMBridge.builder(this.getClass().getName()).withNodes(3).build();
-            cluster = Cluster.builder()
-                    .addContactPoint(CCMBridge.ipOfNode(1))
-                    .withRetryPolicy(Mockito.spy(DowngradingConsistencyRetryPolicy.INSTANCE))
-                    .build();
-            Session session = cluster.connect();
-
-            session.execute("CREATE KEYSPACE test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 3}");
-            session.execute("CREATE TABLE test.foo(k int primary key)");
-            session.execute("INSERT INTO test.foo(k) VALUES (0)");
-
-            // All replicas up: should achieve all levels without downgrading
-            checkAchievedConsistency(ALL, ALL, session);
-            checkAchievedConsistency(QUORUM, QUORUM, session);
-            checkAchievedConsistency(ONE, ONE, session);
-
-            ccm.stop(1);
-            ccm.waitForDown(1);
-            // Two replicas remaining: should downgrade to 2 when CL > 2
-            checkAchievedConsistency(ALL, TWO, session);
-            checkAchievedConsistency(QUORUM, QUORUM, session); // since RF = 3, quorum is still achievable with two nodes
-            checkAchievedConsistency(TWO, TWO, session);
-            checkAchievedConsistency(ONE, ONE, session);
-
-            ccm.stop(2);
-            ccm.waitForDown(2);
-            // One replica remaining: should downgrade to 1 when CL > 1
-            checkAchievedConsistency(ALL, ONE, session);
-            checkAchievedConsistency(QUORUM, ONE, session);
-            checkAchievedConsistency(TWO, ONE, session);
-            checkAchievedConsistency(ONE, ONE, session);
+            query();
+            fail("expected an ReadTimeoutException");
+        } catch (ReadTimeoutException e) {
+            assertThat(e.getConsistencyLevel()).isEqualTo(expectedCL);
```

Comment:
How come this matches? Is it because you passed `received + 1` when priming, and SCassandra uses that to determine the CL it returns in the response?